src/services/codefixes/addMissingAwait.ts

```diff
@@ -0,0 +1,175 @@
+/* @internal */
+namespace ts.codefix {
+    type ContextualTrackChangesFunction = (cb: (changeTracker: textChanges.ChangeTracker) => void) => FileTextChanges[];
+    const fixId = "addMissingAwait";
+    const propertyAccessCode = Diagnostics.Property_0_does_not_exist_on_type_1.code;
+    const callableConstructableErrorCodes = [
+        Diagnostics.This_expression_is_not_callable.code,
+        Diagnostics.This_expression_is_not_constructable.code,
+    ];
+    const errorCodes = [
+        Diagnostics.An_arithmetic_operand_must_be_of_type_any_number_bigint_or_an_enum_type.code,
+        Diagnostics.The_left_hand_side_of_an_arithmetic_operation_must_be_of_type_any_number_bigint_or_an_enum_type.code,
+        Diagnostics.The_right_hand_side_of_an_arithmetic_operation_must_be_of_type_any_number_bigint_or_an_enum_type.code,

```

Comment:
This will show for any old type combination, like `number + Event`, right? What happens in that case? Does it just fail to do anything when you try to "add missing await"?

Reply:
It's supposed to be filtered out further in `getCodeActions` (it doesn’t show anything if you return undefined / empty array in there). This is back to the tradeoffs we were discussing about using related info as a “bonus error message” vs. copying and pasting all these into new, unique messages. I wrote this down as a larger LS design/API change to consider at some later point.